swig/include/csharp/typemaps_csharp.i

```diff
@@ -599,3 +599,19 @@ OPTIONAL_POD(int, int);
 %typemap(imtype) (void* callback_data) "string"
 %typemap(cstype) (void* callback_data) "string"
 %typemap(csin) (void* callback_data) "$csinput"
+
+
+/******************************************************************************
+ * GDALGetNextFeature typemaps                                                *
+ *****************************************************************************/
+
+%apply (double *defaultval) {double* pdfProgressPct};
+
+%typemap(imtype) (OGRLayerShadow **ppoBelongingLayer)  "ref IntPtr"
+%typemap(cstype) (OGRLayerShadow **ppoBelongingLayer) "ref IntPtr"
+%typemap(csin) (OGRLayerShadow **ppoBelongingLayer)  "ref $csinput"
```

Comment:
@andreasbaa I don't know C# at all, but is that sufficient to get a Layer instance from the CSharp method?

Reply:
@rouault Yes, I have done it like this in C#:
`var intPtr = new IntPtr();
double progress = 0;
var feature = ds.GetNextFeature(ref intPtr, ref progress, null, null);
var layer = new Layer(intPtr, false, null);
`

Reply:
hum, that doesn't look very type-safe to me (it looks like you could potentially try to instance another GDAL object that would take a intPtr in its constructor). I guess that ideally we'd want to return a tuple (Feature, Layer), but I've no idea if that's doable in C#, and even less with SWIG. So this is probably a good enough solution for now

Reply:
I totally agree. I tried making SWIG return the Layer object directly, but did not manage to make swig translate the double pointer to a ref Layer. Hopefully someone can contribute to this in the future. Exposing the intpointer makes me at least able to utilize getnextfeature with the gmlas driver :) Big performance increase!